src/sds_data_model/dataframe.py

```diff
@@ -208,47 +209,59 @@ def call_method(
             return return_value
 
 
-def to_zarr(
-    self: Self,
-    path: str,
-    data_array_name: str,
-    index_column_name: str = "bng_index", 
-    geometry_column_name: str = "geometry",
-) -> None:
-    """Reads in data as a Spark DataFrame. A dummy dataset of the BNG is created and written to zarr which is then overwritten 
-    as the Spark DataFrame is converted to a mask then a Dataset.
-    This function assumes that the dataframe contains a column with the BNG 100km grid reference 
-    and a column containing the bounds of the BNG grid refernce system as a list named as "bounds"
-
-    Args:
-        self (Self): _description_
-        path (str): _description_
-        data_array_name (str): _description_
-        index_column_name (str, optional): _description_. Defaults to "bng_index".
-        geometry_column_name (str, optional): _description_. Defaults to "geometry".
-
-    Returns:
-        _type_: _description_
-    """
-
-    _create_dummy_dataset(
-        path=path,
-        data_array_name=data_array_name,
-    )
-    _partial_to_zarr_region = partial(
-        _to_zarr_region,
-        data_array_name=data_array_name,
-        path=path,        
-    )
-    return (
-        self.data
-        .groupby(index_column_name)
-        .applyInPandas(
-            _partial_to_zarr_region,
-            self.data.schema,
+    def to_zarr(
+        self: Self,
```

Comment:
`mypy` doesn't like `Self`, which is a shame, see [here](https://github.com/Defra-Data-Science-Centre-of-Excellence/sds-data-model/pull/153#discussion_r1001639021) for more details.

```suggestion
        self: _DataFrameWrapper,
```

Reply:
Addressed in https://github.com/Defra-Data-Science-Centre-of-Excellence/sds-data-model/pull/160/commits/1ba332df59239e5666ddbef0891e3e0bfb2f6de1